pkg/aws_config_client/completer.go

```diff
@@ -114,49 +114,126 @@ func (c *completer) SurveyProfile() (*AWSNamedProfile, error) {
 		AWSProfile: profile,
 	}
 
-	return namedProfile, nil
+	return []*AWSNamedProfile{namedProfile}, nil
+}
+
+// SurveyRole will ask a user to configure a default role
+func (c *completer) SurveyRoles() ([]*AWSNamedProfile, error) {
+	// first prompt for roles
+	roles := c.awsConfig.GetRoleNames()
+	accounts := c.awsConfig.GetAccounts()
+
+	roleIdx, err := c.prompt.Select(
+		"Select the AWS role you would like to make default:",
+		roles,
+	)
+	if err != nil {
+		return nil, err
+	}
+	targetRole := roles[roleIdx]
+
+	configuredProfiles := []*AWSNamedProfile{}
+
+	for _, account := range accounts {
+		profileName := c.calculateDefaultProfileName(account)
+
+		// get the roles associated with this account
+		profiles := c.awsConfig.GetProfilesForAccount(account)
+		for _, profile := range profiles {
+
```

Comment:
I'd invert this logic a bit. Instead of doing `if profile.RoleName == targetRole` in line 157 you can instead add something lhere like:
```
if profile.RoleName != targetRole {
   continue
}
```
this way we:
- do less work
- early on in the logic make it clear to readers that we only care about profiles named a certain way